### PR TITLE
create simple homepage and results page

### PR DESCRIPTION
**Pull request by AaronLopes**

### CODE REVIEW

File: src/main/webapp/WEB-INF/tags/base.tag [hunk 1]
```diff
@@ -1,4 +1,2 @@
 <%@tag description="Common page content" pageEncoding="UTF-8"%>
-<p>
-  This text appears on every page! Put common headers and navigational elements here.
-</p>
+<h3>CodeSnippet</h3>
```
Comment (kaylaleung) [inline src/main/webapp/WEB-INF/tags/base.tag:2]: Make a separate PR to confirm project name?

File: src/main/webapp/WEB-INF/templates/homepage.jsp [hunk 2]
```diff
@@ -5,15 +5,11 @@
...
   </head>
   <body>
     <t:base/>
-    <p>Hello World!</p>
-    <p>The current date is ${requestScope["date"]}</p>
-    <form method="GET" action="/search">
-      <input name="q" placeholder="Query goes here" required>
-      <input type="submit">
-    </form>
+    <div class="gcse-searchbox-only" data-resultsUrl="/search"></div>
```
Comment (kaylaleung) [inline src/main/webapp/WEB-INF/templates/homepage.jsp:13]: The homepage placement of the search box is identical to the srp placement. Will this be changed to look closer actually Google homepage with a page centered searchbox?

File: src/main/webapp/WEB-INF/templates/homepage.jsp [hunk 3]
```diff
@@ -5,15 +5,11 @@
 <html>
   <head>
     <t:head/>
+    <link href="../searchbox-style.css" rel="stylesheet" type="text/css">
     <title>Project Homepage</title>
```
Comment (irisliu77) [inline src/main/webapp/WEB-INF/templates/homepage.jsp:9]: might need to change to our project name

Comment (irisliu77) [reply src/main/webapp/WEB-INF/templates/homepage.jsp:9]: i mean for the title

File: src/main/webapp/style.css [hunk 4]
```diff
@@ -0,0 +1,7 @@
+h3 {
+  text-align: center;
+}
+
+#results-box {
+  width: 50%;
+}
```
Comment (kaylaleung) [inline src/main/webapp/style.css:7]: Will this width styling be compatible with adding a RHS column on the page with a mock card?

File: src/main/webapp/WEB-INF/templates/homepage.jsp (hunk 2 above)
Comment (AaronLopes) [reply src/main/webapp/WEB-INF/templates/homepage.jsp:13]: For the MVP, I think the unchanging searchbox placement is fine. The placement (along with the searchbox styling) can definitely be changed later to resemble the Google homepage.

File: src/main/webapp/style.css (hunk 4 above)
Comment (AaronLopes) [reply src/main/webapp/style.css:8]: Maybe, but changed to a flexbox bc it'll definitely be easier to work with

File: src/main/webapp/style.css [hunk 5]
```diff
@@ -0,0 +1,8 @@
+h3 {
+  text-align: center;
+}
+
+.srp-container {
+  display: flex;
+  flex-direction: row;
```
Comment (anbenson) [inline src/main/webapp/style.css:7]: This is the default - this is typically left out.

File: src/main/webapp/style.css (hunk 4 above)
Comment (anbenson) [reply src/main/webapp/style.css:8]: It's worth thinking early on about how you want to divide up the width. Is it percentage-based? Hardcoded pixel values? Will there be gaps? Dependent on screen size?

Flexbox definitely works here, but inline-block may be sufficient for your use case as well.

File: src/main/webapp/WEB-INF/templates/homepage.jsp (hunk 2 above)
Comment (anbenson) [reply src/main/webapp/WEB-INF/templates/homepage.jsp:13]: From the design doc I got the impression that there wasn't going to be a homepage anymore. Has that changed?

Also, the design doc's mock's searchbox doesn't extend across the entire page, but that's not super important to address now.

File: src/main/webapp/WEB-INF/tags/head.tag [hunk 6]
```diff
@@ -1,2 +1,3 @@
 <%@tag description="Common head content" pageEncoding="UTF-8"%>
-<!-- This is where common <link> and <script> tags would go. -->
+<link href="../../style.css" rel="stylesheet" type="text/css">
```
Comment (anbenson) [inline src/main/webapp/WEB-INF/tags/head.tag:2]: The href refers to the filepath at _runtime_ (since that's when it's evaluated). At runtime, all the current pages are served from / or /search, so the ../.. is unnecessary. To check what I'm saying, try loading localhost:8080/../../style.css - it'll redirect to localhost:8080/style.css.

File: src/main/webapp/WEB-INF/templates/homepage.jsp [hunk 7]
```diff
@@ -5,15 +5,11 @@
 <html>
   <head>
     <t:head/>
-    <title>Project Homepage</title>
+    <link href="../searchbox-style.css" rel="stylesheet" type="text/css">
```
Comment (anbenson) [inline src/main/webapp/WEB-INF/templates/homepage.jsp:8]: Same thing here - except this file doesn't appear to exist in the PR.

Comment (anbenson) [inline src/main/webapp/WEB-INF/templates/homepage.jsp:8]: type="text/css" is not necessary in HTML5.

File: src/main/webapp/WEB-INF/tags/head.tag [hunk 8]
```diff
@@ -1,2 +1,3 @@
...
+<script async src="https://cse.google.com/cse.js?cx=017022763452466193978:ejddswiptcs"></script>
```
Comment (anbenson) [inline src/main/webapp/WEB-INF/tags/head.tag:3]: Is it safe for this script to be async? I assume it's looking for the gcse-searchbox-only class and so forth, which if the script is fetched asynchronously, you could run into a race condition where the rest of the doc isn't parsed yet.

File: src/main/webapp/style.css (hunk 4 above)
Comment (AaronLopes) [reply src/main/webapp/style.css:8]: Considering that we are displaying two main elements (search results and knowledge cards) the space will be shared 50/50. I want to proceed with flexbox for its (relative) simplicity, as I don't see any need to hardcode width values for now.

File: src/main/webapp/WEB-INF/tags/head.tag [hunk 9]
```diff
@@ -1,2 +1,3 @@
 <%@tag description="Common head content" pageEncoding="UTF-8"%>
-<!-- This is where common <link> and <script> tags would go. -->
+<link href="style.css" rel="stylesheet">
+<script src="https://cse.google.com/cse.js?cx=017022763452466193978:ejddswiptcs"></script>
```
Comment (anbenson) [inline src/main/webapp/WEB-INF/tags/head.tag:3]: With the `async` keyword, my previous comment said there might be a race condition if this script loaded before parsing the rest of the page. Without it, it makes the script synchronously downloaded, making the race condition always occur. But obviously that's not happening, so I took a closer look. It looks like your previous approach with `async` is from the [documentation](https://developers.google.com/custom-search/docs/tutorial/implementingsearchbox), and upon closer inspection, it looks like the script waits for the document to finish loading first, so `async` is fine after all.

Sorry for the back-and-forth.

File: src/main/webapp/style.css (hunk 4 above)
Comment (anbenson) [reply src/main/webapp/style.css:7]: Quick note - style.css is in head.tag, so it'll be included in both the homepage and the srp, but the .srp-container rule is specific to the srp. With the amount of CSS you have currently, it's not going to be an issue to have unused CSS on the homepage, but consider separating out common CSS and page-specific CSS.
